### PR TITLE
[WFCORE-429]: Incremental redeployment (single file update) over management API

### PR DESCRIPTION
**Pull request by ehsavoie**
Adding tests with exploded deployments and overlays.
Adding mixed-domain tests for exploded managed deployments.

This requires a core update with https://github.com/wildfly/wildfly-core/pull/1675 merged.
CI: https://ci.wildfly.org/viewLog.html?buildId=21863&tab=buildResultsDiv&buildTypeId=WF_WildFlyCoreIntegrationExperiments
Jira: https://issues.jboss.org/browse/WFCORE-429

### CODE REVIEW

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/deployment/deploymentoverlay/ExplodedDeploymentOverlayTestCase.java [hunk 1]
```diff
@@ -0,0 +1,246 @@
+package org.jboss.as.test.integration.deployment.deploymentoverlay;
```
Comment (bstansberry) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/deployment/deploymentoverlay/ExplodedDeploymentOverlayTestCase.java]: This needs a copyright header.

**Comment by ehsavoie** [comment]
retest this please

**Comment by ehsavoie** [comment]
retest this please

File: testsuite/mixed-domain/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+Mixed Domain Test Suite
```
Comment (scottmarlow) [inline testsuite/mixed-domain/README.md]: +1000

**Comment by ehsavoie** [comment]
retest this please

**Pull request merged**